### PR TITLE
cargo: fix jemalloc dependencies

### PR DESCRIPTION
**Pull request by nrainer-materialize**
This fixes https://buildkite.com/materialize/nightly/builds/7762#018f744c-16d6-4bde-bbb1-d9e8ffcf2e06.

### CODE REVIEW

**Comment by nrainer-materialize** [comment]
@benesch, is [this](https://github.com/MaterializeInc/materialize/blob/191f02d582f00e1b7a73b6606427346cd10664d5/ci/test/lint-deps/README.md?plain=1#L29) still up to date with recent changes regarding jemalloc?
> ### Specifics for jemalloc
> We enable jemalloc by default on all non-mac platforms, unless --no-default-features is enabled. In this case, jemalloc should be disabled on all platforms. To quickly determine the validity of changes, check that only the file x86_64-unknown-linux-gnu-default mentions the tikv-jemallocator dependency, and all others don't.

**Review by bobbyiliev — APPROVED** [review]

**Review by benesch — CHANGES_REQUESTED** [review]
We need to go the other way with this! As described here, those dependencies are not unused but load bearing for selecting the correct memory allocator by default: https://github.com/MaterializeInc/materialize/pull/27067

I'll push up a revision to this PR that marks them as ignored in the unused dependency list.

**Comment by benesch** [comment]
> @benesch, is [this](https://github.com/MaterializeInc/materialize/blob/191f02d582f00e1b7a73b6606427346cd10664d5/ci/test/lint-deps/README.md?plain=1#L29) still up to date with recent changes regarding jemalloc?

Ah, not quite. Pushed up a revision that updates the text here appropriately.

**Review by benesch — APPROVED** [review]

**Review by antiguru — APPROVED** [review]
LGTM after the most recent changes!

File: ci/test/lint-deps.sh [hunk 1]
```diff
@@ -70,9 +70,9 @@ function deps() {
             --locked \
             --target "$target" \
             "$@" \
-            | tac \
+            | tail -r \
             | awk -F: 'BEGIN {f=-1} /^([0-9]+):'"$crate"'/{print;f=$0-1} f==$1{print;f=f-1}' \
-            | tac)
+            | tail -r)
         # Only print crate and dependency chain if there is a dependency.
```
Comment (antiguru) [inline ci/test/lint-deps.sh:76]: Is `tail -r` really a thing?
```
tail -r
tail: invalid option -- 'r'
Try 'tail --help' for more information.
```

Comment (nrainer-materialize) [reply ci/test/lint-deps.sh:76]: Probably only on macOS? 🤔

**Pull request merged**